### ISSUE TITLE
Crash when re-defining light source (from constantinfinite to infinite)

### ISSUE THREAD

**Issue by Theverat** [LuxCoreRender/LuxCore#238]
Reported by **Weekend**.

Blender and/or graphics driver crash when changing from constantinfinite to infinite light source.

His specs:
> fx8350 stock - 1060 6gb driver 436.02
windows 10 64 bit version 1903 os build 18362.295

My specs:
Windows 10
AMD Ryzen 7 2700x
Nvidia RTX 2080

**Steps to reproduce:**
* Open this .blend file with v2.2RC1 or later: [crash_infinite_light.zip](https://github.com/LuxCoreRender/LuxCore/files/3641801/crash_infinite_light.zip)
* Start a viewport render (upper right corner of the 3D viewport)
* In the "HDRI Settings" panel on the right, select the "studio027.exr" HDRI
* Blender will crash

Full log: https://pastebin.com/BsQ0YnXP
```
[Exporter] Update because of: REQUIRES_SCENE_EDIT | WORLD
Unpacking image "studio027.exr" to temp file "C:\Users\Simon\AppData\Local\Temp\tmpxz64lk20.exr"
[SDL][10.313] Light re-definition: __WORLD_BACKGROUND_LIGHT__
[SDL][10.313] Reading texture map: C:\Users\Simon\AppData\Local\Temp\tmpxz64lk20.exr
[LuxCore][10.375] Compile Lights
[LuxCore][10.406] Lights compilation time: 30ms
[LuxCore][10.406] Compile ImageMaps
[LuxCore][10.406] Image maps page(s) count: 1
[LuxCore][10.422]  RGB channel page 0 size: 23437Kbytes
[LuxCore][10.422] Image maps compilation time: 16ms
Error   : Unrecognized Exception
Address : 0x00007FF83A3EA839
Module  : C:\WINDOWS\System32\KERNELBASE.dll
```

![1](https://user-images.githubusercontent.com/1584761/65420791-53476700-de02-11e9-9e7d-83549fc4b36a.PNG)
![2](https://user-images.githubusercontent.com/1584761/65420795-55a9c100-de02-11e9-8bc8-44cba575f278.PNG)

**Comment by Dade916** [comment]
If you unpack the file "studio027.exr" and than replicate the procedure (selecting the unpacked file), it works fine here (can you confirm ?).

This is a bit strange because I don't see how the two procedures are different from LuxCore point of view (I assume the HDR is a real file in both cases because LuxCore knows nothing of Blender packed data).

Further investigating the problem...

**Comment by Theverat** [comment]
> If you unpack the file "studio027.exr" and than replicate the procedure (selecting the unpacked file), it works fine here (can you confirm ?).

No, I can not confirm this, unpacking doesn't help on my system.

**Comment by Dade916** [comment]
I now realize you are using OpenCL for the view port rendering, right ?

**Comment by Theverat** [comment]
Yes, correct. The problem doesn't appear with RTPathCPU.

**Comment by Dade916** [comment]
I should have fixed the problem.

**Comment by Theverat** [comment]
Confirmed, fixed. Thanks!
